src/ajax/interceptors.cljc

```diff
@@ -170,26 +170,31 @@
                    headers))))
   (-process-response [_ xhrio] xhrio))
 
-(m/defn-curried ^:internal uri-with-params [{:keys [vec-strategy params]} uri]
+(m/defn-curried ^:internal uri-with-params [{:keys [vec-strategy params method url-params]} uri]
   "Internal function. Takes a uri and appends the interpretation of the query string to it
    matching the behaviour of `url-request-format`."
-  (if params
+  (if (and (= method "GET") params)
     (str uri
          (if (re-find #"\?" uri) "&" "?") ; add & if uri contains ?
          (url/params-to-str vec-strategy params))
-    uri))
+    (if url-params
+      (str uri
+           (if (re-find #"\?" uri) "&" "?") ; add & if uri contains ?
+           (url/params-to-str vec-strategy url-params))
+      uri)))
 
 ;;; ProcessGet is one of the standard interceptors
 ;;; Its function is to rewrite the uri of GET requests,
 ;;; since there's no other way to transmit params data
 ;;; in a GET.
-(defrecord ProcessGet []
+(defrecord ProcessUrlParameters []
   Interceptor
   (-process-request [_ {:keys [method] :as request}]
     (if (= method "GET")
```

Comment:
Again, could we get rid of the copy/paste. The following should work in this case `(if (= method "GET") reduced identity)`

Reply:
Cool! Was actually wondering how to do this in Clojure. I fixed it.